v4/core/log.go

```diff
@@ -100,7 +121,9 @@ func (l *SDKLoggerImpl) Debug(format string, inserts ...interface{}) {
 func NewLogger(level LogLevel, stdLogger *log.Logger) *SDKLoggerImpl {
 	return &SDKLoggerImpl{
 		logLevel: level,
-		goLogger: stdLogger,
+		// if a user provides their own logger, use it everywhere
+		goLoggerStdout: stdLogger,
+		goLoggerStderr: stdLogger,
```

Comment:
I didn't want to make a breaking change by editing the signature of `NewLogger`. If you think I should, or if you think I should implement a second "constructor" that allows a user to provide both a Stdout and a Stderr logger, let me know